xbmc/windowing/X11/GLContextEGL.cpp

```diff
@@ -321,7 +329,9 @@ EGLConfig CGLContextEGL::getEGLConfig(EGLDisplay eglDisplay, XVisualInfo *vInfo)
     }
     if (value == (EGLint)vInfo->visualid) {
       eglConfig = eglConfigs[i];
-      break;
+      //Don't select the first one, let's keep the last instead
+      //eg. SGX 545 on X11 returns 3 configurations with the same visualid but the first has depth=0
```

Comment:
I think we need something more robust here. How do you know that this does not break other platforms?